generator/main.ts

```diff
@@ -23,6 +23,33 @@ import * as webidl from 'webidl2';
 
 const TEMPLATE_DIR = path.resolve(__dirname, '../../../template');
 
+async function generateBacardiInterface(
+    env: nunjucks.Environment, input_idl_path: string, output_path: string) {
+  const parsedData =
+      webidl.parse(await file.read(path.resolve(input_idl_path)));
+  const idl_interface: idls.Interface = new idls.InterfaceImpl(parsedData[0]);
+
+  const [header_tmpl, cpp_tmpl] = await Promise.all([
+    file.read(path.resolve(TEMPLATE_DIR, 'bacardi_header.njk')),
+    file.read(path.resolve(TEMPLATE_DIR, 'bacardi_cpp.njk'))
+  ]);
+  const idl_name = input_idl_path.replace('.idl', '').replace('.', '');
+
+  const header_file_path = path.resolve(
+      output_path,
+      path.dirname(idl_name) + '/' +
```

Comment:
if we were not use "example" folder for generated file's path, this line was not needed any more.